### PR TITLE
[msal-node] documentation

### PR DESCRIPTION
**Pull request by sameerag**
Add readme and docs for msal-node

### CODE REVIEW

File: lib/msal-common/docs/Response.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MSAL will return an `AuthenticationResult.ts` object as a response to all public request APIs:
```
Comment (sangonzal) [inline lib/msal-common/docs/Response.md:1]: "to all acquire token APIs". (doesn't apply to get accounts, remove accounts etc)

File: lib/msal-node/README.md [hunk 2]
```diff
@@ -1,50 +1,154 @@
...
+
+- [Authorization Code Grant](https://oauth.net/2/grant-types/authorization-code/) with [PKCE](https://oauth.net/2/pkce/)
+- [Device Code Grant](https://oauth.net/2/grant-types/device-code/)
+- [Refresh Token Grant](https://oauth.net/2/grant-types/refresh-token/)
+- Silent Flow (user convenience flow to acquire a token silently)
+
+In the upcoming quarters we plan to add support for:
+
+- Client Credentials flow
+- Confidential client flow (authorization code flow with client secret)
```
Comment (sangonzal) [inline lib/msal-node/README.md:38]: I would say. 

```
Confidential client flows: 
- Authorization code flow with client secret
- Client credentials flow
- On-behalf-of flow
```

File: lib/msal-node/README.md [hunk 3]
```diff
@@ -1,50 +1,154 @@
...
+
+More details on different grant types supported by Microsoft authentication libraries in general can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows)
+
+### App models supported:
+
+The app models supported with this library are:
```
Comment (sangonzal) [inline lib/msal-node/README.md:44]: "The scenarios". Also on line 50.

File: lib/msal-node/README.md [hunk 4]
```diff
@@ -1,50 +1,154 @@
...
+| ------| ------- | ---------| --------- |
+| July 13th, 2020 (Tentative) | @azure/msal-node v1.0.0-alpha.1 | No release notes yet | Full version of the `@azure/msal-node` package; relies on `@azure/msal-common` v1.0.0 |
+| July 6th, 2020  | @azure/msal-node v1.0.0-alpha.0| No release notes yet | Full version of the `@azure/msal-node` package; relies on `@azure/msal-common` v1.0.0-beta.4 |
+
+## Prerequisites
+
+Before using `@azure/msal-node` you will need to register your app in the azure portal to get a valid `clientId` for configuration, and to register the routes that your app will accept redirect traffic on if applicable. Currently we support the below app registrations for `@azure/msal-node`:
+
+- [Desktop app that calls web APIs: App registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-app-registration)
+- [Protected web API: App registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-protected-web-api-app-registration)
```
Comment (sangonzal) [inline lib/msal-node/README.md:70]: I think this is for protecting APIs (which we don't support)

File: lib/msal-node/README.md [hunk 5]
```diff
@@ -1,50 +1,154 @@
...
+npm run build
+
+// Change to the msal-node package directory
+cd lib/msal-node/
+
+// To run build only for node package
+npm run build
 
+// to link msal-node and msal-common packages
+lerna bootstrap
```
Comment (sangonzal) [inline lib/msal-node/README.md:117]: lerna bootstrap should be the first thing.

File: lib/msal-node/docs/configuration.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+## Usage
+
+The configuration object can be passed into the `PublicClientApplication` constructor. The only required config parameter is the `client_id` of the application. Everything else is optional, but may be required depending on your authentication flow, tenant and application model.
+
+[Configuration](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-node/modules/_src_config_configuration_.html#configuration) object with all supported parameters is as below:
+
+```javascript
+
+// Call back API tp read from the cache in .json format
```
Comment (sangonzal) [inline lib/msal-node/docs/configuration.md:19]: to*

File: lib/msal-node/docs/initialize-public-client-application.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+```javascript
+import * as msal from "@azure/msal-node";
+
+const publicClientConfig = {
+    auth: {
+        clientId: "your_client_id",
+        authority: "your_authority",
+        // mandatory only for authorization code flow
+        redirectUri: "your_redirect_uri",
```
Comment (sangonzal) [inline lib/msal-node/docs/initialize-public-client-application.md:23]: I think node doesn't have redirectUri as part of public client config. It's only there for the auth code requests.

File: lib/msal-node/docs/request.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Request
+
+Since MSAL Node supports various authorization code grants, there is support for different public APIs per grant and the corresponding request.
+
+## Authorization Code Flow
+
+### Public APIs
+- [getAuthCodeUrl()](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-node/classes/_src_client_publicclientapplication_.publicclientapplication.html#getauthcodeurl): This API is the first leg of the `authorization code grant` for MSAL Node. The request is of the type [AuthorizationUrlRequest](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-common/modules/_src_request_authorizationurlrequest_.html).
+The application is sent a URL that can be used to generate an `authorization code` by any browser of choice, and then proceed to call the below API for the `token acquisition` .
```
Comment (sangonzal) [inline lib/msal-node/docs/request.md:9]: "can be opened in a browser of choice, where the user can input their credential, and will be redirected back to the redirect URI registered with an authorization code. The authorization code can be redeemed for a token..."

File: lib/msal-node/docs/request.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+- [acquireTokenByCode()](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-node/classes/_src_client_publicclientapplication_.publicclientapplication.html#acquiretokenbycode): This API is the second leg of the `authorization code grant` for MSAL Node. The request constructed here should be of the type [AuthorizationCodeRequest](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-common/modules/_src_request_authorizationcoderequest_.html). The application passed the `authorization code` received as a part of the above step and exchanges it for a `token`.
+
+``` javascript
+
+    const authCodeUrlParameters = {
+        scopes: ["user.read"],
+        redirectUri: ["http://localhost:3000/redirect"],
```
Comment (sangonzal) [inline lib/msal-node/docs/request.md:17]: This should be a string.

File: lib/msal-common/docs/Response.md (hunk 1 above)
Comment (sameerag) [reply lib/msal-common/docs/Response.md:1]: Changed it.

File: lib/msal-node/README.md (hunk 2 above)
Comment (sameerag) [reply lib/msal-node/README.md:38]: Done

File: lib/msal-node/README.md (hunk 3 above)
Comment (sameerag) [reply lib/msal-node/README.md:44]: Done

File: lib/msal-node/README.md (hunk 4 above)
Comment (sameerag) [reply lib/msal-node/README.md:70]: Yes. I think it is confidential flow. Thanks.

File: lib/msal-node/README.md (hunk 5 above)
Comment (sameerag) [reply lib/msal-node/README.md:117]: Done

File: lib/msal-node/docs/configuration.md (hunk 6 above)
Comment (sameerag) [reply lib/msal-node/docs/configuration.md:19]: Corrected.

File: lib/msal-node/docs/initialize-public-client-application.md (hunk 7 above)
Comment (sameerag) [reply lib/msal-node/docs/initialize-public-client-application.md:23]: You are right. Done.

File: lib/msal-node/docs/request.md (hunk 8 above)
Comment (sameerag) [reply lib/msal-node/docs/request.md]: Done.

File: lib/msal-node/docs/request.md (hunk 9 above)
Comment (sameerag) [reply lib/msal-node/docs/request.md]: Done

**Pull request merged**